pkg/apis/helm.cattle.io/v1/types.go

```diff
@@ -57,4 +57,5 @@ type HelmChartConfig struct {
 type HelmChartConfigSpec struct {
 	ValuesContent string `json:"valuesContent,omitempty"`
 	FailurePolicy string `json:"failurePolicy,omitempty"`
+	BackOffLimit  int32  `json:"backOffLimit,omitempty"`
```

Comment:
I feel like this belongs in the core HelmChart, not in the HelmChartConfig.

The idea is that the core HelmChart is the primary driver of behavior; the HelmChartConfig has only the bare minimum required to allow end users to customize the values or behavior for HelmCharts packaged with a Kubernetes distribution and installed by this controller.

This value isn't something we would want to have users overriding for any of the charts packaged with K3s or RKE2, so it belongs in the HelmChart only.

Reply:
sounds good. I will make the changes.